### PR TITLE
Use dnspython to check specifically for an "MX" DNS record

### LINKED ISSUE #14829: Users with some special email domains can not create an account - domain check fails
**Describe the bug**

Users with the email domain `@edu.unito.it` can not create an account. Instead, they get the error message `The email domain cannot be resolved.` which is created here with a check of the domain exists.

https://github.com/galaxyproject/galaxy/blob/895e028cff83db0ad20f05c3cb87ef6c1b443241/lib/galaxy/security/validate_user_input.py#L103

It seems the domain EDU.unito.it does not exist, `unito.it` does.

**Galaxy Version and/or server at which you observed the bug**
Galaxy Version: 22.05.1

**To Reproduce**
1. Go to the login page
2. try to create an account with `foobar@edu.unito.it`

**Expected behavior**

I can create an account.

### PR DESCRIPTION
**Pull request by nsoranzo**
when validating the domain name of the email address for a new user.

Fix https://github.com/galaxyproject/galaxy/issues/14829 .

## How to test the changes?
(Select all options that apply)
- [ ] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [x] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT).

### CODE REVIEW

File: lib/galaxy/security/validate_user_input.py [hunk 1]
```diff
@@ -97,11 +99,11 @@ def validate_email(trans, email, user=None, check_dup=True, allow_empty=False, v
     return message
 
 
-def validate_domain_resolves(domain):
+def validate_email_domain_name(domain: str) -> LiteralString:
     message = ""
     try:
-        socket.gethostbyname(domain)
-    except socket.gaierror:
+        dns.resolver.resolve(domain, "MX")
```
Comment (martenson) [inline lib/galaxy/security/validate_user_input.py:105]: According to the spec when there is no MX record the address resolution is supposed to fall back on the A record.

> If an empty list of MXs is returned,
> the address is treated as if it was associated with an implicit MX
> RR, with a preference of 0, pointing to that host. 

source: https://www.rfc-editor.org/rfc/rfc5321#section-5.1

Comment (nsoranzo) [reply lib/galaxy/security/validate_user_input.py:105]: Maybe dnspython is smart enough to deal with such a situation? We'd need a test case.

Comment (blankenberg) [reply lib/galaxy/security/validate_user_input.py:105]: FQDNs. MX first and overriding, CNAME second and only with lack of MX, then queue, else error -- no other fall backs?

Comment (nsoranzo) [reply lib/galaxy/security/validate_user_input.py:105]: I've randomly found a test case, a domain that has an A record but not MX: example.it

@martenson I've added a fall back to the A record that works for the test case.

@blankenberg That's a bit too cryptic for me, sorry! Feel free to suggest code changes or push to my branch!

**Review by martenson — APPROVED** [review]

**Pull request merged**